### ISSUE TITLE
Multiple Push Notifications

### ISSUE THREAD

**Issue by siddhartpai** [phonegap/phonegap-plugin-push#396]
Hi I'm receiving the same notification multiple times on my app. Is this a bug ?

**Comment by robaweb** [comment]
+1 same problem! 
ios cases:
1) In looked screen notification receiving once, it's ok
2) In background mode i receiving the same notification twice. That is bug. Fix needed
3) In foreground mode notifications receiving once, ok

**Comment by robaweb** [comment]
looks like in ios 9.2 bug is fixed for my case
@siddhartpai can you check is please ?

**Comment by macdonst** [comment]
@siddhartpai probably not. My guess is you have multiple `on('notification')` event handlers.

**Comment by slowsword** [comment]
In 9.2 there is still the bug.

I compile the app and run from xcode (onto device), then go to home screen, then send push. I get 2 notifications within a second or two. 

If i lock the screen and send push, i only get the appropriate 1 notification. If I unlock and am at home screen, send push, i only get the one notification as expected. This is while it is still debugging. If i stop debuggin and i get the 2 notifications when on home screen, and only one notification when locked. Locking and unlocking does not seem to make it notify once, only when its debugging does that temporarily fix it.

I only have 1 on('notification'). 

Long story short, bug still exists on iOS 9.2 and 9.2.1

**Comment by macdonst** [comment]
I cannot reproduce this on an iOS 9.2 device. Need more info here folks in order to reproduce this.

**Comment by kenvunz** [comment]
```

**Comment by kenvunz** [comment]
I have the exact same issue, but re-install the app again seem to solve it

Found this on [stackoverflow](http://stackoverflow.com/a/33935224), might worth a try if the problem persists

> From iOS 9 every time when you uninstall and than re-install the app again a new device token has assigned this might be the reason that you receives the multiple push notifications.

> Actually I read out from one forum, they provide the solution that when you generating a payload that add one extra custom any random value so that each payload has some unique value. in my case in vb.net I am using DateTime.Now.ToString("MMddyyyyHHmmssfff") to add a unique time stamp with milliseconds. I hope its work I implemented this but not tested so far.

**Comment by macdonst** [comment]
Closing for inactivity
